ecs-agent/netlib/platform/common_linux.go

```diff
@@ -0,0 +1,200 @@
+	"github.com/aws/amazon-ecs-agent/ecs-agent/netlib/model/networkinterface"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/netlib/model/tasknetworkconfig"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/pkg/errors"
+)
+
+const (
+	// Identifiers for each platform we support.
+	WarmpoolDebugPlatform    = "ec2-debug-warmpool"
+	FirecrackerDebugPlatform = "ec2-debug-firecracker"
+	WarmpoolPlatform         = "warmpool"
+	FirecrackerPlatform      = "firecracker"
+)
+
+// common will be embedded within every implementation of the platform API.
+// It contains all fields and methods that can be commonly used by all
+// platforms.
+type common struct {
+	nsUtil ecscni.NetNSUtil
+}
+
+// NewPlatform creates an implementation of the platform API depending on the
+// platform type where the agent is executing.
+func NewPlatform(
+	platformString string,
+	nsUtil ecscni.NetNSUtil) (API, error) {
+	commonPlatform := common{
+		nsUtil: nsUtil,
+	}
+
+	switch platformString {
+	case WarmpoolPlatform:
+		return &containerd{
+			common: commonPlatform,
+		}, nil
+	}
+	return nil, errors.New("invalid platform: " + platformString)
+}
+
+// BuildTaskNetworkConfiguration translates network data in task payload sent by ACS
+// into the task network configuration data structure internal to the agent.
+func (c *common) BuildTaskNetworkConfiguration(
+	taskID string,
+	taskPayload *ecsacs.Task) (*tasknetworkconfig.TaskNetworkConfig, error) {
+	mode := aws.StringValue(taskPayload.NetworkMode)
+	var netNSs []*tasknetworkconfig.NetworkNamespace
+	var err error
+	switch mode {
+	case ecs.NetworkModeAwsvpc:
+		netNSs, err = c.buildAWSVPCNetworkNamespaces(taskID, taskPayload)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to translate network configuration")
+		}
+	case ecs.NetworkModeBridge:
+		return nil, errors.New("not implemented")
+	case ecs.NetworkModeHost:
+		return nil, errors.New("not implemented")
+	case ecs.NetworkModeNone:
+		return nil, errors.New("not implemented")
+	default:
+		return nil, errors.New("invalid network mode: " + mode)
+	}
+
+	return &tasknetworkconfig.TaskNetworkConfig{
+		NetworkNamespaces: netNSs,
+		NetworkMode:       mode,
+	}, nil
+}
+
+func (c *common) GetNetNSPath(netNSName string) string {
+	return c.nsUtil.GetNetNSPath(netNSName)
+}
+
+// buildAWSVPCNetworkNamespaces returns list of NetworkNamespace which will be used to
+// create the task's network configuration. All cases except those for FoF is covered by
+// this method. FoF requires a separate specific implementation because the network setup
+// is different due to the presence of the microVM.
+// Use cases covered by this method are:
+//  1. Single interface, network namespace (the only externally available config).
+//  2. Single netns, multiple interfaces (For a non-managed multi-ENI experience. Eg EKS use case).
+//  3. Multiple netns, multiple interfaces (future use case for internal customer who need
```

Comment:
Seems this 3rd case is not implemented yet?

Reply:
The method accounts for the third use case as well. I have included a unit test for the use case in the `network_builder_linux_test.go` file. The test case name is "containerd-multi-netns".